packages/chain-adapters/src/cosmossdk/types.ts

```diff
@@ -88,4 +88,9 @@ export type FeeData = {
   gasLimit: string
 }
 
+export type CosmosSDKToken = {
```

Comment:
I wonder if this belongs here or even needs its own type at all?

It looks like it's only going to be used in the context of building an osmosis tx.  Sure maybe it could be used for cosmos related code in the future but we don't really know that.

My 2 cents is leaving it out of the cosmos types reduces complexity and spagetification by a very small amount (adding to 1 type file is nicer than 2)

But its so small it really doesnt matter. just a thought

Reply:
@elmutt I see where you're coming from. I did it this way because personally I think that 
```
export type BuildLPAddTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenInMaxs: [cosmossdk.CosmosSDKToken, cosmossdk.CosmosSDKToken]
}

export type BuildLPRemoveTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenOutMins: [cosmossdk.CosmosSDKToken, cosmossdk.CosmosSDKToken]
}
```

is more readable than 
```
export type BuildLPAddTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenInMaxs: [{ denom: string; amount: string }, { denom: string; amount: string }]
}

export type BuildLPRemoveTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenOutMins: [{ denom: string; amount: string }, { denom: string; amount: string }]
}
```

This is more about making the code self-documenting than it is about future reusability of the type, but that's definitely an option should we add more Cosmos zones in the near future. This is a small thing, so maybe let's just leave it for now.

Reply:
@elmutt I see where you're coming from. I did it this way because personally I think that 
```
export type BuildLPAddTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenInMaxs: [cosmossdk.CosmosSDKToken, cosmossdk.CosmosSDKToken]
}

export type BuildLPRemoveTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenOutMins: [cosmossdk.CosmosSDKToken, cosmossdk.CosmosSDKToken]
}
```

is more readable than 
```
export type BuildLPAddTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenInMaxs: [{ denom: string; amount: string }, { denom: string; amount: string }]
}

export type BuildLPRemoveTxInput<T extends ChainId> = BuildLPTxInput<T> & {
  tokenOutMins: [{ denom: string; amount: string }, { denom: string; amount: string }]
}
```

This is more about making the code self-documenting than it is about future reusability of the type, but that's definitely an option should we add more Cosmos zones in the near future. This is a small thing, so maybe let's just leave it for now.

Reply:
I prefer types for objects as you can also use them when construction the input payload if needed.

Reply:
+1, LGTM @pastaghost.